commcare_connect/opportunity/tables.py

```diff
@@ -52,6 +56,41 @@ def show_warning(record):
     return ""
 
 
+class UserVisitReviewFilter(FilterSet):
+    review_status = ChoiceFilter(choices=VisitReviewStatus.choices, empty_label="All Reviews")
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.form.helper = FormHelper()
+        self.form.helper.disable_csrf = True
+        self.form.helper.form_class = "form-inline"
+        self.form.helper.layout = Layout(Row(Column("review_status", css_class="col-md-3")))
+
+        self.form.fields["review_status"].widget.attrs.update({"@change": "$refs.reviewFilterForm.submit()"})
+
+    class Meta:
+        model = UserVisit
+        fields = ["review_status"]
+
+
+class UserVisitFilter(UserVisitReviewFilter):
+    status = ChoiceFilter(choices=VisitValidationStatus.choices, empty_label="All Visits")
+
+    def __init__(self, *args, **kwargs):
+        managed_opportunity = kwargs.pop("managed_opportunity", False)
+        super().__init__(*args, **kwargs)
+        fields = ["status"]
+        if managed_opportunity:
+            fields.append("review_status")
+        self.form.helper.layout = Layout(Row(*[Column(field, css_class="col-md-3") for field in fields]))
+        for field in fields:
+            self.form.fields[field].widget.attrs.update({"@change": "$refs.visitFilterForm.submit()"})
```

Comment:
This could be added in the form level attrs instead of for each field. Again, if you use the base template, that is already there. See above comment.